File: apps/AI-Keyboard/iOS/Shared/Tone.swift
import Foundation

/// Rewrite tones offered as chips under the Rewrite action.
enum Tone: String, Codable, CaseIterable, Identifiable, Hashable {
    case professional
    case casual
    case friendly
    case romantic

    var id: String { rawValue }

    /// Chip label.
    var label: String {
        switch self {
        case .professional: return "Professional"
        case .casual:       return "Casual"
        case .friendly:     return "Friendly"
        case .romantic:     return "Romantic"
        }
    }

    /// Phrase injected into the rewrite prompt.
    var descriptor: String {
        switch self {
        case .professional: return "polished and professional"
        case .casual:       return "relaxed and casual"
        case .friendly:     return "warm and friendly"
        case .romantic:     return "affectionate and romantic"
        }
    }
}

File: apps/AI-Keyboard/Android/app/src/main/java/ai/zetic/demo/cherrypad/model/Tone.kt
package ai.zetic.demo.cherrypad.model

/** Rewrite tones. `descriptor` is injected into the rewrite prompt. */
enum class Tone(val label: String, val descriptor: String) {
    PROFESSIONAL("Professional", "polished and professional"),
    CASUAL("Casual", "relaxed and casual"),
    FRIENDLY("Friendly", "warm and friendly"),
    ROMANTIC("Romantic", "affectionate and romantic"),
}
